nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,584 @@
+                "bottom": {eye: {x: 0, y: 0, z: -1.7},
+                           up: {x: 0, y: 1, z: 0},
+                           center: {x: 0, y: 0, z: 0}},
+                "front": {eye: {x: 0, y: 1.7, z: 0},
+                          up: {x: 0, y: 0, z: 1},
+                          center: {x: 0, y: 0, z: 0}},
+                "back": {eye: {x: 0, y: -1.7, z: 0},
+                         up: {x: 0, y: 0, z: 1},
+                         center: {x: 0, y: 0, z: 0}},
+            };
+
+            return cameras[view];
+
+        }
+
+
+        function makePlot(surface, hemisphere, divId) {
+
+            info = surfaceMapInfo[surface + "_" + hemisphere];
+
+            info["type"] = "mesh3d";
+
+
+            for (let attribute of ["x", "y", "z"]) {
+                if (!(attribute in info)) {
+                    info[attribute] = decodeBase64(
+                        info["_" + attribute], "float32");
+                }
+            }
+
+            for (let attribute of ["i", "j", "k"]) {
+                if (!(attribute in info)) {
+                    info[attribute] = decodeBase64(
+                        info["_" + attribute], "int32");
+                }
+            }
+
+            info["vertexcolor"] = surfaceMapInfo["vertexcolor_" + hemisphere];
+
+            let data = [info];
+
+            info['lighting'] = getLighting();
+            let layout = getLayout();
+            let config = getConfig();
+
+            Plotly.react(divId, data, layout, config);
+
+            addColorbar(divId, layout, config);
+
+        }
+    </script>
+    <script>
+        $(document).ready(
+            function() {
+                if(surfaceMapInfo["full_brain_mesh"]){
+                    $("#select-hemisphere").change(addPlot);
+                    $("#select-kind").change(addPlot);
+                }
+                else{
+                    $("#select-hemisphere").hide();
+                    $("#select-kind").hide();
+                }
+                addPlot();
+                $("#select-view").change(updateLayout);
+                $("#surface-plot").mouseup(function() {
+                    $("#select-view").val("custom");
+                });
+                $(window).resize(updateLayout);
+
+            });
+    </script>
+</head>
+
+<body>
+    <div id="surface-plot"></div>
+    <select id="select-hemisphere">
+<option value="left">Left hemisphere</option>
+<option value="right">Right hemisphere</option>
+</select>
+
+    <select id="select-kind">
+<option value="inflated">Inflated</option>
+<option value="pial">Pial</option>
+</select>
+    <select id="select-view">
+<option value="left">view: Left</option>
+<option value="right">view: Right</option>
+<option value="front">view: Front</option>
+<option value="back">view: Back</option>
+<option value="top">view: Top</option>
+<option value="bottom">view: Bottom</option>
+<option value="custom">view: -</option>
+</select>
+
+</body>
+
+</html>
+
+"""
```

Comment:
I think that it would be good if this string was written in a separate file, ending with ".html". It simplifies reading the code (ie syntax highlighting).

Reply:
done. note that it isn't valid html though, only a template that needs to be filled